consumer/pubsub/src/main/java/com/spotify/heroic/consumer/pubsub/PubSubConsumerModule.java

```diff
@@ -0,0 +1,321 @@
+        @Named("consuming")
+        @PubSubScope
+        AtomicInteger consuming() {
+            return new AtomicInteger();
+        }
+
+        @Provides
+        @Named("total")
+        @PubSubScope
+        AtomicInteger total() {
+            return new AtomicInteger();
+        }
+
+        @Provides
+        @Named("errors")
+        @PubSubScope
+        AtomicLong errors() {
+            return new AtomicLong();
+        }
+
+        @Provides
+        @Named("consumed")
+        @PubSubScope
+        LongAdder consumed() {
+            return new LongAdder();
+        }
+
+        @Provides
+        @Named("subscriptionId")
+        @PubSubScope
+        String subscriptionId() { return subscriptionId; }
+
+        @Provides
+        @Named("topicId")
+        @PubSubScope
+        String topicId() { return topicId; }
+
+        @Provides
+        @Named("projectId")
+        @PubSubScope
+        String projectId() { return projectId; }
+
+        @Provides
+        @Named("maxOutstandingElementCount")
+        @PubSubScope
+        Long maxOutstandingElementCount() { return maxOutstandingElementCount; }
+
+        @Provides
+        @Named("maxOutstandingRequestBytes")
+        @PubSubScope
+        Long maxOutstandingRequestBytes() { return maxOutstandingRequestBytes; }
+
+        @Provides
+        @PubSubScope
+        ConsumerSchema.Consumer consumer(final IngestionManager ingestionManager) {
+            final IngestionGroup ingestion = ingestionManager.useDefaultGroup();
+
+            if (ingestion.isEmpty()) {
+                throw new IllegalStateException("No backends are part of the ingestion group");
+            }
+
+            final ConsumerSchema.Depends dep = DaggerConsumerSchema_Depends
+                .builder()
+                .primaryComponent(primary)
+                .depends(depends)
+                .dependsModule(new ConsumerSchema.DependsModule(ingestion))
+                .build();
+            final ConsumerSchema.Exposed exposed = schema.setup(dep);
+            return exposed.consumer();
+        }
+
+        @Provides
+        @PubSubScope
+        public Managed<Connection> connection(
+            final AsyncFramework async,
+            final ConsumerReporter reporter,
+            final ConsumerSchema.Consumer consumer,
+            @Named("maxOutstandingElementCount") Long maxOutstandingElementCount,
+            @Named("maxOutstandingRequestBytes") Long maxOutstandingRequestBytes,
+            @Named("consuming") AtomicInteger consuming,
+            @Named("total") AtomicInteger total,
+            @Named("errors") AtomicLong errors,
+            @Named("consumed") LongAdder consumed
+        ) {
+            return async.managed(new ManagedSetup<Connection>() {
+                @Override
+                public AsyncFuture<Connection> construct() {
+                    return async.call(() -> {
+                        log.info("p:{}, t:{}, s:{}", projectId, topicId, subscriptionId);
```

Comment:
Just for clarity, can we change the log text to read "projectId: {projectId}, topicId:{topicId}, subscriptionId:{subscriptionId}"

Reply:
:+1:

Reply:
That doesn't seem to work with slf4j.